webapp/src/containers/createIncident/index.tsx

```diff
@@ -0,0 +1,314 @@
+        }
+
+        setApiError(error);
+        setShowResultPanel(true);
+    };
+
+    useEffect(() => {
+        const {isLoading, isError, isSuccess, error, data} = getIncidentState();
+        setShowModalLoader(isLoading);
+        if (isError && error) {
+            handleError(error);
+        }
+
+        if (isSuccess) {
+            setApiError(null);
+            if (!showChannelPanel) {
+                setShowResultPanel(true);
+                return;
+            }
+
+            const subscriptionEvents = [
+                SubscriptionEvents.STATE,
+                SubscriptionEvents.PRIORITY,
+                SubscriptionEvents.COMMENTED,
+                SubscriptionEvents.ASSIGNMENT_GROUP,
+                SubscriptionEvents.ASSIGNED_TO,
+            ];
+
+            const payload: CreateSubscriptionPayload = {
+                server_url: SiteURL ?? '',
+                is_active: true,
+                user_id: Cookies.get(Constants.MMUSERID) ?? '',
+                type: SubscriptionType.RECORD,
+                record_type: RecordType.INCIDENT,
+                record_id: data.sys_id || '',
+                subscription_events: subscriptionEvents.join(','),
+                channel_id: channel ?? currentChannelId,
+            };
+
+            setSubscriptionPayload(payload);
+            makeApiRequest(Constants.pluginApiServiceConfigs.createSubscription.apiServiceName, payload);
+        }
+    }, [getIncidentState().isError, getIncidentState().isSuccess, getIncidentState().isLoading]);
+
+    useEffect(() => {
+        if (subscriptionPayload) {
+            const {isLoading, isError, isSuccess, error} = getSubscriptionState();
+            setShowModalLoader(isLoading);
+            if (isError && error) {
+                handleError(error);
+            }
+
+            if (isSuccess) {
+                setApiError(null);
+                dispatch(refetch());
+                setShowResultPanel(true);
+            }
+        }
+    }, [getSubscriptionState().isError, getSubscriptionState().isSuccess, getSubscriptionState().isLoading]);
+
+    useEffect(() => {
+        if (currentChannelId) {
+            setChannel(currentChannelId);
+        }
+
+        if (open && getGlobalModalState(pluginState).data) {
+            const {description: reduxStateDescription, senderId: reduxSenderId} = getGlobalModalState(pluginState).data as IncidentModalData;
+            setSenderId(reduxSenderId);
+            if (reduxStateDescription.length > Constants.MaxShortDescriptionLimit) {
+                setDescription(reduxStateDescription);
+            } else if (reduxStateDescription.length > Constants.MaxShortDescriptionCharactersView) {
+                setShortDescription(reduxStateDescription.slice(0, Constants.MaxShortDescriptionCharactersView) + '...');
+            } else {
+                setShortDescription(reduxStateDescription);
+            }
+        }
+    }, [open]);
```

Comment:
data and currentChannelId?

Reply:
If we change the current channel the modal will close and the data here is post data, I think we should re-open the modal after updating the post